exonum/src/events/codec.rs

```diff
@@ -38,79 +39,49 @@ impl MessagesCodec {
 }
 
 impl Decoder for MessagesCodec {
-    type Item = RawMessage;
+    type Item = Vec<u8>;
     type Error = failure::Error;
 
     fn decode(&mut self, buf: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
-        // Read header
-        if buf.len() < HEADER_LENGTH {
+        // Framing level
+        if buf.len() < mem::size_of::<u32>() {
             return Ok(None);
         }
 
         let len = LittleEndian::read_u32(buf) as usize;
 
-        if buf.len() < len + NOISE_HEADER_LENGTH {
+        if buf.len() < NOISE_HEADER_LENGTH + len {
             return Ok(None);
         }
 
-        let mut buf = self.session.decrypt_msg(len, buf)?;
+        let buf = self.session.decrypt_msg(len, buf)?;
 
-        if buf[0] != 0 {
-            bail!("A first byte of the message must be set to 0");
-        }
-
-        // Check payload len
-        let total_len = LittleEndian::read_u32(&buf[6..10]) as usize;
-
-        if total_len as u32 > self.max_message_len {
+        if buf.len() > self.max_message_len as usize {
             bail!(
-                "Received message is too long: {}, maximum allowed length is {} bytes",
-                total_len,
-                self.max_message_len,
-            );
-        }
-
-        if total_len < HEADER_LENGTH {
-            bail!(
-                "Received malicious message with insufficient \
-                 size in header: {}, expected header size {}",
-                total_len,
-                HEADER_LENGTH
-            );
+                "Message too big received_len = {}, allowed_len = {}",
```

Comment:
I suggest to write it as `Received message is too long: received_len = {}, allowed_len = {}`, so it's grammatically correct